mitodl/mit-open
# Upgrade to opensearch 2.11

# What are the relevant tickets?
Fixes #263 
Related: https://github.com/mitodl/ol-infrastructure/issues/1959

# Description (What does it do?)
- Changes the docker-compose opensearch image version to 2.11.1
- Updates the code to fix breaking changes



# How can this be tested?
- Run `docker compose build --no-cache web celery`
- Run `docker compose up`
- Run `docker compose run --rm web python manage.py recreate_index --all`.  It should complete without errors
- Try one or two backpopulate commands (with relevant .env values populated), they should complete without errors:
  ```
  ./manage.py backpopulate_xpro_data
  ./manage.py backpopulate_podcast_data
  ```
- Try out searches with various parameters, they should return expected results. For example:
  http://localhost:8063/api/v1/learning_resources_search/?platform=xpro
  http://localhost:8063/api/v1/learning_resources_search/?resource_type=podcast
  http://localhost:8063/api/v1/learning_resources_search/?resource_type=course&sortby=mitcoursenumber
  http://localhost:8063/api/v1/learning_resources_search/?department=8&sortby=mitcoursenumber

- Run `docker compose run --rm python manage.py update_index --all`, it should complete without errors.
